src/helpers/dom.js

```diff
@@ -42,9 +42,9 @@ export const findActiveNode = (possibleNodes, isScrollingDown) => {
     if (!x.current) {
       return false;
     }
-    const { bottom } = x.current.getBoundingClientRect();
+    const { top, bottom } = x.current.getBoundingClientRect();
     return isScrollingDown
-      ? bottom > topEdge && bottom < window.innerHeight
-      : bottom < bottomEdge && bottom > 0;
+      ? top < bottomEdge && top > 0
+      : bottom > topEdge && bottom < window.innerHeight;
```

Comment:
Relevant change to the "active section" tracking logic. Basically the rest of this PR is because after changing these lines, the side nav and URL got out of sync sometimes.